### PR TITLE
Feat/reintroduce secreted status

### PR DESCRIPTION
**Pull request by pouneh**
## Motivation

Currently the API responses include a "secreted" when listing balances. However, this field was unhelpfully always set to zero. Between removing the API, and reintroducing the status, we reintroduce the status.

## In this PR

We introduce 2 new txo status.

    "Created" - for output TXOs that are proposed as part of a built transaction (either not submitted, or failed)
    "Secreted" - for output TXOS that are build as part of successfully submitted transactions that go to a wallet that does NOT belong to the sender

### CODE REVIEW

File: full-service/src/json_rpc/v1/models/txo.rs [hunk 1]
```diff
@@ -66,6 +66,7 @@ impl From<&db::txo::TxoStatus> for TxoStatus {
             db::txo::TxoStatus::Spent => TxoStatus::Spent,
             db::txo::TxoStatus::Unspent => TxoStatus::Unspent,
             db::txo::TxoStatus::Unverified => TxoStatus::Unspent,
+            db::txo::TxoStatus::Created | db::txo::TxoStatus::Secreted => todo!(),
```
Comment (briancorbin) [inline full-service/src/json_rpc/v1/models/txo.rs:69]: How come this is left as todo?

File: tools/build-fs.sh [hunk 2]
```diff
@@ -90,7 +90,7 @@ case ${net} in
         if test -f "${WORK_DIR}/consensus-enclave.css"; then
             CONSENSUS_ENCLAVE_CSS="${WORK_DIR}/consensus-enclave.css"
         else
-            CONSENSUS_ENCLAVE_CSS=`pwd`/target/docker/release/consensus-enclave.css
+            CONSENSUS_ENCLAVE_CSS=`pwd`/mobilecoin/target/docker/release/consensus-enclave.css
```
Comment (briancorbin) [inline tools/build-fs.sh:93]: Can we move these changes to a separate PR? Out of scope for this review

File: tools/run-fs.sh [hunk 3]
```diff
@@ -130,7 +130,7 @@ case "${net}" in
         if [[ -z "${offline}" ]]
         then
             MC_PEER="insecure-mc://localhost:3200/,insecure-mc://localhost:3201/"
-            MC_TX_SOURCE_URL="http://localhost:4566/node-0-ledger/,http://localhost:4566/node-1-ledger/"
+            MC_TX_SOURCE_URL="file:///home/$USER/full-service/mobilecoin/target/docker/release/mc-local-network/node-ledger-distribution-0,file:///home/$USER/full-service/mobilecoin/target/docker/release/mc-local-network/node-ledger-distribution-1"
```
Comment (briancorbin) [inline tools/run-fs.sh:133]: Can we move these changes to a separate PR? Out of scope for this review

File: tools/test.sh [hunk 4]
```diff
@@ -4,6 +4,12 @@
 
 # RUSTFLAGS="-C instrument-coverage" \
 
+
+if ! [[ -f "$(pwd)/consensus-enclave.css" ]]; then
+  echo "Please copy the consensus enclave to $(pwd)"
+  exit 1
+fi
+
```
Comment (briancorbin) [inline tools/test.sh:12]: Can we move these changes to a separate PR? Out of scope for this review

**Review by briancorbin — CHANGES_REQUESTED** [review]
This looks good so far, but it could use a bunch more tests, both directly in the db/txo.rs file and in the e2e tests. I would recommend creating tests for a variety of possible situations that the database could be in to ensure we get the correct status at the correct times.

It would also be a good idea to test that querying an individual txo results in the correct status, again both in the e2e tests as well as the db/txo.rs tests

**Review by christian-oudard — APPROVED** [review]

**Review by briancorbin — APPROVED** [review]
This looks good, thanks for all of the changes!

**Pull request merged**